catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/UpdateOperations.java

```diff
@@ -733,4 +736,37 @@ private UpdateStorageRequest applyAttributeOverrides(UpdateStorageRequest update
                 metacardMap);
         return updateStorageRequest;
     }
+
+    private UpdateRequest populateResourceUris(UpdateRequest updateRequest, QueryResponse query) {
+        Set<Metacard> updateRequestMetacards = updateRequest.getUpdates()
+                .stream()
+                .map(Map.Entry::getValue)
+                .filter(metacard -> metacard.getResourceURI() == null)
```

Comment:
I would prefer to move away from these convenience methods in favor of getting attributes by their constants instead, as this results in more consistent code and API usage.

Also, I don't think checking for null is sufficient here. You probably want `StringUtils.isEmpty/isNotEmpty` instead.